packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { TimeConstants, addMinutes, formatTimeString, ceilMinuteToIncrement } from '@fluentui/date-time-utilities';
+import { ComboBox, IComboBox, IComboBoxOption } from '../../ComboBox';
+import { ITimePickerProps, ITimeRange } from './TimePicker.types';
+
+// Valid KeyChars for user input
+const KEYCHAR_0 = 48;
+const KEYCHAR_COLON = 58;
+const KEYCHAR_SPACE = 32;
+const KEYCHAR_A = 65;
+const KEYCHAR_M = 77;
+const KEYCHAR_P = 80;
+
+export const TimePicker = ({
+  label,
+  increments = 30,
+  showSeconds = false,
+  allowFreeform = true,
+  useHour12 = false,
+  timeRange = { start: -1, end: -1 },
+  onFormatDate,
+  onValidateUserInput,
+  ...rest
+}: ITimePickerProps) => {
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>(0);
+  const [userText, setUserText] = React.useState<string>('');
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+
+  const defaultTime = generateDefaultTime(increments, timeRange);
+  const optionsCount = getDropdownOptionsCount(increments, timeRange);
+  const timePickerOptions: IComboBoxOption[] = Array(optionsCount)
+    .fill(0)
+    .map((_, index) => {
+      const option = addMinutes(defaultTime, increments * index);
+      option.setSeconds(0);
+      return {
+        key: index,
+        text: onFormatDate ? onFormatDate(option) : `${formatTimeString(option, showSeconds, useHour12)}`,
+      };
+    });
+
+  const onChange = React.useCallback(
+    (event: React.FormEvent<IComboBox>, option?: IComboBoxOption, index?: number, value?: string): void => {
+      const validateUserInput = (userInput: string): string => {
+        let errorMessageToDisplay = '';
+        let regexString;
```

Comment:
Nit, just call this `regex` (it's not a string) and declare the type as `RegExp` upfront (right now it will be `any`)

Reply:
Resolved by renaming to `regex` and declaring type as `RegExp`.